core/node/state_keeper/src/keeper.rs

```diff
@@ -582,34 +593,51 @@ impl ZkSyncStateKeeper {
                     "L1 batch #{} should be sealed unconditionally as per sealing rules",
                     updates_manager.l1_batch.number
                 );
+
+                // Push the current block if it has not been done yet
+                if is_last_block_sealed {
+                    self.io
+                        .update_next_l2_block_timestamp(updates_manager.next_l2_block_timestamp());
+                    tracing::debug!(
+                        "Initialized new L2 block #{} (L1 batch #{}) with timestamp {}",
+                        updates_manager.l2_block.number + 1,
+                        updates_manager.l1_batch.number,
+                        display_timestamp(updates_manager.next_l2_block_param.timestamp)
+                    );
+                    Self::start_next_l2_block(updates_manager, batch_executor).await?;
+                }
                 return Ok(());
             }
 
-            if self.io.should_seal_l2_block(updates_manager) {
+            if self.io.should_seal_l2_block(updates_manager) && !is_last_block_sealed {
```

Comment:
Does ordering of the clauses matter here (I guess from the perspective of `should_seal_l2_block` side-effects)? Logically, it makes sense to reorder the clauses, so that the cheaper one (`!is_last_block_sealed`) is checked first.